src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -559,33 +568,26 @@ private JButton getSaveButton() {
 						medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
```

Comment:
Please replace lines 520:539 with:

```
					inventory.setChargeType(chargeType == null ? null : chargeType.getCode());
					inventory.setDischargeType(dischargeType == null ? null : dischargeType.getCode());
					inventory.setSupplier(supplier == null ? null : supplier.getSupId());
					inventory.setDestination(destination == null ? null : destination.getCode());
```

and similar changes in `mode.equals("update") { .. }` block, lines 584:607